melusine/nlp_tools/lemmatizer.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+from typing import Union
+
+import numpy
+import spacy
+from spacy.language import Language
+from spacy_lefff import LefffLemmatizer, POSTagger, Downloader
+
+from melusine.utils.verbitim_transformer import VerbitimTransformer
+
+
+@Language.factory('melt_tagger')
+def create_melt_tagger(nlp, name):
+    # TODO:Variable de release de l'asset
+    return POSTagger()
+
+
+@Language.factory('lefff_lemmatizer')
+def create_french_lemmatizer(nlp, name):
+    return LefffLemmatizer(after_melt=True, default=True)
+
+
+class BadConfigurationError(Exception):
+    """Exception to raise when a bad configuration was passed to lemmatization engine"""
+    pass
+
+
+class Lemmatizer(VerbitimTransformer):
+    """ Lemmatizer component for Socle NLP
+        Available engines include : spacy, Lefff and TreeTagger
+    """
+
+    _AVAILABLE_ENGINES = ['spacy',
+                          'Lefff'
+                          ]
+
+    _SPACY_ARGS = {'exclude': ['senter', 'ner']}
+
+    def __init__(self,
+                 in_col: str,
+                 out_col: str,
+                 engine: str,
+                 engine_conf: Union[dict, str, None],
+                 n_jobs: int = 1,
+                 batch_size: int = 1000
+                 ):
+        """
+        Initialize a Socle NLP Lemmatizer object
+        :param in_col:  input column of dataframe.
+        :param out_col: output column, will be created if it does not exist.
+        :param engine: lemmatization engine to load.
+        :param engine_conf: configuration for the engine chosen. For TreeTagger, a dictionary containing TAGDIR,
+                            TAGLANG, TAGPARFILE keys is required. For spacy based lemmatizers, the name of the model
+                            must be provided. Refer to tutorials for more information.
+        :param n_jobs: spacy-based engines only. Number of processes to be spawned by spacy pipe.
+        :param batch_size: spacy-based engines only. Batch size to process at once in spacy pipe.
+        """
+
+        self.in_col = in_col
+        self.out_col = out_col
+        self.engine = engine
+        self.engine_conf = engine_conf
+        self.n_jobs = n_jobs
+        self.batch_size = batch_size
+
+        self.logger = logging.getLogger(__name__)
```

Comment:
We do not recommand adding logger into the object but preferably at a module level.